src/contracts/Staking.sol

```diff
@@ -207,6 +223,35 @@ contract Staking is Ownable {
         return tokePoolContract.balanceOf(address(this)); // TODO: verify pending withdraws are a part of this
     }
 
+    /**
+        @notice checks TOKE's cycleTime is withink duration to batch the transactions
+        @return bool
+     */
+    function canBatchTransactions() internal view returns (bool) {
+        ITokeManager iTokeManager = ITokeManager(tokeManager);
+        uint256 offset = 50; // amount of blocks before the next cycle to batch the withdrawal requests
```

Comment:
How do we update this when their cycles change?

Reply:
the offset is just the amount of blocks before their next cycle.  We get the cycle start time and duration directly from the toke manager contract.  Is this what you were asking?

Reply:
yea, I am asking when they change their cycles, I assume we need to update this offset, right?

Reply:
I dm'd you about this, but I'm going to go ahead and merge this as it's the first of the 3 PR's that are available.  If we need to make changes we can